src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -672,6 +678,49 @@ def ok_to_stop(self,
             return HandleCommandResult(-errno.EBUSY, '', warn_message)
         return HandleCommandResult(0, warn_message, '')
 
+    def find_prometheus_credentials(self, multicluster_config: Dict[str, Any], url: str) -> Optional[Dict[str, Any]]:
+        for _, clusters in multicluster_config['config'].items():
+            for cluster in clusters:
+                prometheus_url = cluster['prometheus_url']
```

Comment:
use `cluster.get('prometheus_url', '')` in case the key doesn't exist this code will fail with an exception..

Reply:
Updated